web/preferences.js

```diff
@@ -157,37 +109,10 @@ class BasePreferences {
       throw new Error("Please use `about:config` to change preferences.");
     }
     await this.#initializedPromise;
-    const defaultValue = this.#defaults[name],
-      oldPrefs = structuredClone(this.#prefs);
+    AppOptions.setAll({ [name]: value }, /* prefs = */ true);
```

Comment:
Similar to the question above, is `[name]` correct here? I would assume just `{ name: value }` and don't really see why the name should be an array here, but maybe I'm just confused about the expected input format for `AppOptions.setAll` here. Perhaps it would help if we document in a docstring at `AppOptions.setAll` what kind of format the `options` argument should have?

Reply:
It's not actually an array, but rather a way to create an Object where the key becomes whatever *is in* the `name` variable rather than using the "name" string. Try running e.g. the following code in the console:
```javascript
var name = "annotationEditorMode", value = -1;
console.log({ [name]: value });
```
and note how the output is `{ annotationEditorMode: -1 }`

There's more examples in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#computed_property_names

Reply:
D'oh, I think I was mixing up JavaScript and Python here. Thanks for clarifying!